src/transformers/generation/utils.py

```diff
@@ -663,7 +662,8 @@ def _update_model_kwargs_for_generation(
                     dim=-1,
                 )
 
-        model_kwargs["cache_position"] = model_inputs.get("cache_position", None)
+        if "cache_position" in model_kwargs and model_kwargs["cache_position"] is not None:
+            model_kwargs["cache_position"] = model_kwargs["cache_position"][-1:] + 1
```

Comment:
my single worry here is potential stride, adding a `.contiguous()` might be needed

Reply:
I've double-checked, it's always `(1,)` 🤗 (which makes sense, since it's a 1D tensor)

Its shape will indeed be different, at least between prefill and subsequent generation